src/main/java/picard/vcf/AccumulateVariantCallingMetrics.java

```diff
@@ -88,11 +88,14 @@ protected int doWork() {
                     detailedMetrics.calculateFromDerivedFields();
                     totalHetDepth += detailedMetrics.TOTAL_HET_DEPTH;
 
-                    // add it to the list of metrics for that sample so that we can merge them later
-                    sampleDetailsMap.computeIfAbsent(detailedMetrics.SAMPLE_ALIAS, f -> new ArrayList<>()).add(detailedMetrics);
+                    // collapse it into to the main metrics for that sample
+                    final CollectVariantCallingMetrics.VariantCallingDetailMetrics sampleDetails = collapsedSampleDetailsMap.getOrDefault(detailedMetrics.SAMPLE_ALIAS, new CollectVariantCallingMetrics.VariantCallingDetailMetrics());
+                    CollectVariantCallingMetrics.VariantCallingDetailMetrics.foldInto(sampleDetails, Collections.singleton(detailedMetrics));
+                    sampleDetails.calculateDerivedFields();
+                    collapsedSampleDetailsMap.put(detailedMetrics.SAMPLE_ALIAS, sampleDetails);
```

Comment:
is it required to do this? the object is already in the map.

Reply:
Yes, you still need to do this.  It's never set otherwise so if you don't put it then it just maintains the default empty collection (I confirmed this with local tests).

Reply:
right. because you're using getOrDefault()...use computeIfAbsent instead and you will be able to avoid the 20K*(number of shards) put() calls.